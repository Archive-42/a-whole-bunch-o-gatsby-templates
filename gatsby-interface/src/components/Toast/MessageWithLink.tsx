/** @jsx jsx */
import React from "react"
import { css, jsx } from "@emotion/core"
import { Link } from "../Link"

const messageCss = css`
  margin-bottom: 8px;
`

export interface MessageWithLinkProps {
  linkLabel: React.ReactNode
  href?: string
  target: string
  to?: string
}

export const MessageWithLink: React.FC<MessageWithLinkProps> = ({
  children,
  linkLabel,
  ...linkProps
}) => (
  <React.Fragment>
    <div css={messageCss}>{children}</div>
    <Link
      // we need to cast to "any" because TS claims that "href" is required for <Link /> even though it's not
      {...(linkProps as any)}
    >
      {linkLabel}
    </Link>
  </React.Fragment>
)
