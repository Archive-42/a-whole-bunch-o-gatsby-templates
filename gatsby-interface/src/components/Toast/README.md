## Toast

A set of tools to display toast messages.

```javascript
import {
  ToastProvider,
  useShowSuccessToast,
  useShowErrorToast,
} from "gatsby-interface"

// Somewehere at the top of your React app tree
function App({ children }) {
  return <ToastProvider>{children}</ToastProvider>
}

function Component() {
  const showSuccessToast = useShowSuccessToast()
  const showErrorToast = useShowErrorToast()

  return (
    <>
      <button onClick={() => showSuccessToast("Your action was successful!")}>
        Show success toast
      </button>
      <button
        onClick={() =>
          showSuccessToast("This message will stay on screen until closed", {
            timeout: 0,
          })
        }
      >
        Show and persist success toast
      </button>
      <button onClick={() => showErrorToast("An error occured!")}>
        Show error toast
      </button>
    </>
  )
}
```

### Components

#### ToastProvider

This component is required to display toasts. Render it somewhere at the top of your React app tree.

```javascript
import { ToastProvider } from "gatsby-interface"

// Somewehere at the top of your React app tree
function App({ children }) {
  return <ToastProvider>{children}</ToastProvider>
}
```

#### ToastConsumer

Use this component to get access to `showToast` function if you can't use hooks for some reason.

```javascript
import { ToastConsumer } from "gatsby-interface"

function Component() {
  return (
    <ToastConsumer>
      {({ showToast }) => (
        <button
          onClick={() => showToast(`Your action was successful`, { tone: 'SUCCESS' })}
        >
          Show toast
        </Button>
      )}
    </ToastConsumer>
  );
}
```

### Hooks

#### useShowSuccessToast

Should be used to show a toast that indicates that some action has been successful. The toast will disappear automatically after 5 seconds.

```javascript
import { useShowSuccessToast } from "gatsby-interface"

function Component() {
  const showSuccessToast = useShowSuccessToast()
  const showErrorToast = useShowErrorToast()

  return (
    <button onClick={() => showSuccessToast("Your action was successful!")}>
      Show success toast
    </button>
  )
}
```

#### useShowErrorToast

Should be used to show a toast that indicates that some action has failed. The toast WILL NOT disappear automatically.

```javascript
import { useShowErrorToast } from "gatsby-interface"

function Component() {
  const showErrorToast = useShowErrorToast()

  return (
    <button onClick={() => showErrorToast("An error occured!")}>
      Show error toast
    </button>
  )
}
```

#### useShowErrorAlert

Allows to show an error toast similar to `useShowErrorToast`; the only difference is that you can provide a link to be displayed in the toast message.

```javascript
import { useShowErrorAlert } from "gatsby-interface"

function Component() {
  const showErrorAlert = useShowErrorAlert()

  return (
    <button
      onClick={() =>
        showErrorAlert("An error occured!", {
          to: "/some-path",
          linkLabel: "See details",
        })
      }
    >
      Show error toast
    </button>
  )
}
```

Similar to [Link](../Link), you can pass either `to` or `href` prop as well as `target`.

#### useShowToast

This is a low-level hook and generally should not be used if you can use one of the previous hooks.

```javascript
import { useShowToast } from "gatsby-interface"

function Component() {
  const showToast = useShowToast()

  return (
    <>
      <button
        onClick={() =>
          showToast("Your action was successful!", { tone: "SUCCESS" })
        }
      >
        Show success toast
      </button>
      <button
        onClick={() =>
          showToast("An error occured!", { tone: "DANGER", timeout: 0 })
        }
      >
        Show error toast
      </button>
    </>
  )
}
```
