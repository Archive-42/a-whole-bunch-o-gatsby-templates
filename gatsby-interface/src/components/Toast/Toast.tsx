/** @jsx jsx */
import React from "react"
import Alert from "@reach/alert"
import { keyframes, css, jsx } from "@emotion/core"
import { MdDone, MdClose, MdWarning } from "react-icons/md"

import { ToastTone } from "./types"
import fontSizes from "../../theme/fontSizes"
import dimensions from "../../theme/dimensions"
import space from "../../theme/space"
import colors from "../../theme/colors"
import radii from "../../theme/radii"

const toastEntryAnimation = keyframes`
  100% {
     transform: perspective(1000px) rotateX(0);
  }
`

const toastCss = css`
  align-items: center;
  animation: ${toastEntryAnimation} 0.5s 0.25s ease forwards;
  background: ${colors.grey[90]};
  border-left: 8px solid ${colors.green[50]};
  border-radius: ${radii[2]} ${radii[2]} 0 0;
  color: ${colors.green[5]};
  display: flex;
  font-size: ${fontSizes[1]};
  min-height: ${dimensions.toast.minHeight};
  max-width: calc(100% - (${space[7]} * 2));
  padding-left: ${space[4]};
  transform: perspective(1000px) rotateX(90deg);
  transform-origin: bottom center;

  svg {
    height: auto;
    width: calc(${dimensions.toast.minHeight} * 0.4);
  }

  &:not(:first-of-type) {
    border-radius: ${radii[2]};
    margin-bottom: ${space[1]};
  }
`

const messageCss = css`
  line-height: 1;
  margin: 0 ${space[2]} 0 ${space[3]};
`

const statusCss = css`
  align-items: center;
  color: ${colors.green[50]};
  display: flex;
`

const closeButtonCss = css`
  align-items: center;
  background: none;
  border: none;
  color: ${colors.grey[40]};
  cursor: pointer;
  display: flex;
  height: ${dimensions.toast.minHeight};
  justify-content: center;
  width: ${dimensions.toast.minHeight};
`

const toastColorByTone = {
  SUCCESS: colors.green[50],
  DANGER: colors.red[60],
}

const ToastIconByTone = {
  SUCCESS: MdDone,
  DANGER: MdWarning,
}

export interface ToastProps {
  message: React.ReactNode
  onClose: () => void
  closeButtonLabel: string
  tone: ToastTone
}

export const Toast: React.FC<ToastProps> = ({
  message,
  tone,
  closeButtonLabel,
  onClose,
}) => {
  const IconComponent = ToastIconByTone[tone]

  return (
    <Alert
      css={[
        toastCss,
        css({
          borderLeftColor: toastColorByTone[tone],
        }),
      ]}
      data-testid="toast"
      type={tone === `DANGER` ? `assertive` : `polite`}
    >
      <span
        css={[
          statusCss,
          css({
            color: toastColorByTone[tone],
          }),
        ]}
      >
        <IconComponent />
      </span>
      <div css={messageCss}>{message}</div>
      <button
        css={closeButtonCss}
        type="button"
        onClick={onClose}
        aria-label={closeButtonLabel}
      >
        <MdClose />
      </button>
    </Alert>
  )
}
