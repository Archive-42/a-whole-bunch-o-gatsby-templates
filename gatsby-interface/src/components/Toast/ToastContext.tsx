import React from "react"
import { ToastTone } from "./types"

export interface ToastOptions {
  tone?: ToastTone
  timeout: number
}

export interface ToastContextDefinition {
  showToast: (message: React.ReactNode, toastArg?: ToastOptions) => void
}

export const ToastContext = React.createContext<ToastContextDefinition>({
  showToast: () => undefined,
})

export function useToastContext() {
  const context = React.useContext(ToastContext)

  if (!context) {
    throw new Error(
      `Toast hooks cannot be used outside the ToastProvider component`
    )
  }

  return context
}
