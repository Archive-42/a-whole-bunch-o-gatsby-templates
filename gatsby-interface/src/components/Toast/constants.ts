import { ToastTone } from "./types"

export const DEFAULT_TIMEOUT = 5000
export const DEFAULT_TONE: ToastTone = `SUCCESS`
