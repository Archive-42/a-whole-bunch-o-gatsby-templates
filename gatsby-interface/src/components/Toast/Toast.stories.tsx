import React from "react"
import { storiesOf } from "@storybook/react"
import { StoryUtils } from "../../utils/storybook"
import { Button } from "../Button"
import {
  ToastProvider,
  ToastConsumer,
  useShowErrorToast,
  useShowSuccessToast,
  useShowErrorAlert,
} from "."
import README from "./README.md"

storiesOf(`Toast`, module)
  .addParameters({
    options: {
      showPanel: true,
    },
    readme: {
      sidebar: README,
    },
  })
  .add(`Default`, () => {
    function ErrorToastExample() {
      const showErrorToast = useShowErrorToast()

      return (
        <Button onClick={() => showErrorToast(`An error occured`)}>
          Show error toast
        </Button>
      )
    }
    function ErrorAlertExample() {
      const showErrorAlert = useShowErrorAlert()

      return (
        <Button
          onClick={() =>
            showErrorAlert(`An error occured`, {
              href: `https://google.com`,
              linkLabel: `See details`,
              target: `_blank`,
            })
          }
        >
          Show error alert
        </Button>
      )
    }
    function NoTimeoutExample() {
      const showSuccessToast = useShowSuccessToast()

      return (
        <Button
          onClick={() =>
            showSuccessToast(`This message will stay on screen until closed`, {
              timeout: 0,
            })
          }
        >
          Show toast without auto hide
        </Button>
      )
    }
    return (
      <StoryUtils.Container>
        <StoryUtils.Stack>
          <ToastProvider>
            <ToastConsumer>
              {({ showToast }) => (
                <React.Fragment>
                  <Button
                    onClick={() => showToast(`Your action was successful`)}
                  >
                    Show toast
                  </Button>
                </React.Fragment>
              )}
            </ToastConsumer>
            <NoTimeoutExample />
            <ErrorToastExample />
            <ErrorAlertExample />
          </ToastProvider>
        </StoryUtils.Stack>
      </StoryUtils.Container>
    )
  })
