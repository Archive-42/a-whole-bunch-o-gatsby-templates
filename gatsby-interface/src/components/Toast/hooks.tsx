import React, { useState, useRef, useCallback } from "react"
import { useToastContext } from "./ToastContext"
import { MessageWithLink } from "./MessageWithLink"
import { DEFAULT_TIMEOUT, DEFAULT_TONE } from "./constants"
import { ToastTone } from "./types"

export const useShowToast = () => {
  const { showToast } = useToastContext()

  return showToast
}

export const useShowSuccessToast = () => {
  const showToast = useShowToast()

  return useCallback(
    (message, options = {}) => {
      showToast(message, { ...options, tone: `SUCCESS` })
    },
    [showToast]
  )
}

export const useShowErrorToast = () => {
  const showToast = useShowToast()

  return useCallback(
    (message, options = {}) => {
      showToast(message, { ...options, tone: `DANGER`, timeout: 0 })
    },
    [showToast]
  )
}

export const useShowErrorAlert = () => {
  const showToast = useShowErrorToast()

  return useCallback(
    (message, linkProps, options = {}) => {
      showToast(
        <MessageWithLink {...linkProps}>{message}</MessageWithLink>,
        options
      )
    },
    [showToast]
  )
}

export interface Toast {
  id: symbol
  message: React.ReactNode
  tone: ToastTone
}

export const useToastActions = () => {
  const [toasts, setToasts] = useState<Toast[]>([])
  const timeoutsRef = useRef<Map<symbol, number>>(new Map())

  const removeToast = useCallback((toastId: symbol) => {
    setToasts(prevToasts => prevToasts.filter(({ id }) => id !== toastId))

    window.clearTimeout(timeoutsRef.current.get(toastId))

    timeoutsRef.current.delete(toastId)
  }, [])

  const showToast = useCallback(
    (message, { tone = DEFAULT_TONE, timeout = DEFAULT_TIMEOUT } = {}) => {
      const toastId = Symbol(`toast`)

      setToasts(prevToasts => [...prevToasts, { id: toastId, message, tone }])

      if (timeout > 0) {
        const timeOutId = window.setTimeout(() => {
          removeToast(toastId)
        }, timeout)

        timeoutsRef.current.set(toastId, timeOutId)
      }
    },
    []
  )

  return { toasts, showToast, removeToast }
}
