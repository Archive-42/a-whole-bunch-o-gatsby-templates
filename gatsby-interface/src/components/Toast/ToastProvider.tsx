/** @jsx jsx */
import { css, jsx } from "@emotion/core"
import React from "react"
import { Toast } from "./Toast"
import { ToastContext } from "./ToastContext"
import { useToastActions } from "./hooks"
import zIndices from "../../theme/zIndices"

export const ToastConsumer = ToastContext.Consumer

const containerCss = css`
  align-items: center;
  bottom: 0;
  display: flex;
  flex-direction: column-reverse;
  left: 50%;
  position: fixed;
  transform: translate(-50%, 0);
  width: 100%;
  z-index: ${zIndices.base};
`

export interface ToastProviderProps {
  closeButtonLabel?: string
}

export const ToastProvider: React.FC<ToastProviderProps> = ({
  children,
  closeButtonLabel = `Close`,
}) => {
  const { toasts, showToast, removeToast } = useToastActions()

  const contextValue = React.useMemo(() => {
    return {
      showToast,
    }
  }, [showToast])

  return (
    <ToastContext.Provider value={contextValue}>
      {children}

      <div css={containerCss}>
        {toasts.map((toast, index) => (
          <Toast
            key={`toast-${index}`} // this is probably not the best, but we can't use symbol as key :(
            {...toast}
            onClose={() => removeToast(toast.id)}
            closeButtonLabel={closeButtonLabel || "close"}
          />
        ))}
      </div>
    </ToastContext.Provider>
  )
}
