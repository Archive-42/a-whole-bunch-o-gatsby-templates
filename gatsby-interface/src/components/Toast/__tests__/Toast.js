import React from "react"
import { render, fireEvent, act } from "@testing-library/react"
import {
  ToastProvider,
  ToastConsumer,
  useShowSuccessToast,
  useShowErrorToast,
  useShowErrorAlert,
} from "../"
import { Toast } from "../Toast"

describe(`Toast`, () => {
  const baseProps = {
    message: `Lorem ipsum`,
    onClose: () => {
      // do nothing.
    },
    closeButtonLabel: "close",
    tone: `SUCCESS`,
    id: 0,
  }

  it(`renders unchanged`, async () => {
    const { container } = render(
      <React.Fragment>
        <Toast {...baseProps} />
        <Toast {...baseProps} id={1} tone="DANGER" />
      </React.Fragment>
    )

    expect(container).toMatchSnapshot()
  })

  it(`displays the toast message`, async () => {
    const { queryByText } = render(<Toast {...baseProps} />)

    expect(queryByText(`Lorem ipsum`)).toBeTruthy()
  })

  it(`has accessible label for the close button`, async () => {
    const { container } = render(<Toast {...baseProps} />)

    expect(container.querySelector(`button`)).toHaveAttribute(
      `aria-label`,
      `close`
    )
  })

  it(`calls onRemove callback when clicking the close button`, async () => {
    const removeFn = jest.fn()
    const { getByLabelText } = render(
      <Toast {...baseProps} onClose={removeFn} />
    )

    fireEvent.click(getByLabelText(`close`))

    expect(removeFn).toHaveBeenCalled()
  })
})

describe(`ToastProvider`, () => {
  it(`exposes toast tools via context`, () => {
    const renderFn = jest.fn()

    render(
      <ToastProvider>
        <ToastConsumer>{renderFn}</ToastConsumer>
      </ToastProvider>
    )

    expect(renderFn).toHaveBeenCalledWith(
      expect.objectContaining({
        showToast: expect.any(Function),
      })
    )
  })

  function TestComponent({ toastOptions = {} }) {
    return (
      <ToastConsumer>
        {({ showToast }) => (
          <button onClick={() => showToast(`Lorem ipsum`, toastOptions)}>
            Show toast
          </button>
        )}
      </ToastConsumer>
    )
  }

  it(`allows to specify custom label for the toast close button`, () => {
    const { getByText, getByTestId } = render(
      <ToastProvider closeButtonLabel="Zamknąć">
        <TestComponent toastOptions={{ timeout: 0 }} />
      </ToastProvider>
    )

    act(() => {
      fireEvent.click(getByText(`Show toast`))
    })

    expect(getByTestId(`toast`).querySelector(`button`)).toHaveAttribute(
      `aria-label`,
      `Zamknąć`
    )
  })

  it(`removes a toast on timeout`, () => {
    const { getByText, queryByTestId } = render(
      <ToastProvider>
        <TestComponent toastOptions={{ timeout: 500 }} />
      </ToastProvider>
    )

    act(() => {
      fireEvent.click(getByText(`Show toast`))
    })

    expect(queryByTestId(`toast`)).toBeTruthy()

    setTimeout(() => expect(queryByTestId(`toast`)).toBeFalsy(), 500)
  })

  it(`persists a toast if the provided timeout is 0`, () => {
    const { getByText, queryByTestId } = render(
      <ToastProvider>
        <TestComponent toastOptions={{ timeout: 0 }} />
      </ToastProvider>
    )

    act(() => {
      fireEvent.click(getByText(`Show toast`))
    })

    expect(queryByTestId(`toast`)).toBeTruthy()

    setTimeout(() => expect(queryByTestId(`toast`)).toBeTruthy(), 600)
  })
})

describe(`useShowSuccessToast hook`, () => {
  it(`returns a method to show success toast`, () => {
    function TestComponent() {
      const showSuccessToast = useShowSuccessToast()

      return (
        <button onClick={() => showSuccessToast(`Lorem ipsum`)}>
          Show toast
        </button>
      )
    }

    const { getByText, queryByTestId } = render(
      <ToastProvider>
        <TestComponent />
      </ToastProvider>
    )

    act(() => {
      fireEvent.click(getByText(`Show toast`))
    })

    expect(queryByTestId(`toast`)).toBeTruthy()
  })
})

describe(`useShowErrorToast hook`, () => {
  it(`returns a method to show error toast`, () => {
    function TestComponent() {
      const showErrorToast = useShowErrorToast()

      return (
        <button onClick={() => showErrorToast(`Lorem ipsum`)}>
          Show toast
        </button>
      )
    }

    const { getByText, queryByTestId } = render(
      <ToastProvider>
        <TestComponent />
      </ToastProvider>
    )

    act(() => {
      fireEvent.click(getByText(`Show toast`))
    })

    expect(queryByTestId(`toast`)).toBeTruthy()
    setTimeout(() => expect(queryByTestId(`toast`)).toBeTruthy(), 600)
  })
})

describe(`useShowErrorAlert hook`, () => {
  it(`returns a method to show error alert`, () => {
    function TestComponent() {
      const showErrorAlert = useShowErrorAlert()

      return (
        <button
          onClick={() =>
            showErrorAlert(`Lorem ipsum`, {
              linkLabel: `See details`,
              href: `https://google.com`,
              target: `_blank`,
            })
          }
        >
          Show toast
        </button>
      )
    }

    const { getByText, queryByTestId } = render(
      <ToastProvider>
        <TestComponent />
      </ToastProvider>
    )

    act(() => {
      fireEvent.click(getByText(`Show toast`))
    })

    expect(queryByTestId(`toast`)).toBeTruthy()
    setTimeout(() => expect(queryByTestId(`toast`)).toBeTruthy(), 600)

    const link = getByText(`See details`)
    expect(link).toHaveAttribute(`href`, `https://google.com`)
    expect(link).toHaveAttribute(`target`, `_blank`)
  })

  it(`allows to show error alert with an internal link`, () => {
    function TestComponent() {
      const showErrorAlert = useShowErrorAlert()

      return (
        <button
          onClick={() =>
            showErrorAlert(`Lorem ipsum`, {
              linkLabel: `See details`,
              to: `/local`,
              target: `_blank`,
            })
          }
        >
          Show toast
        </button>
      )
    }

    const { getByText } = render(
      <ToastProvider>
        <TestComponent />
      </ToastProvider>
    )

    act(() => {
      fireEvent.click(getByText(`Show toast`))
    })

    const link = getByText(`See details`)
    expect(link).toHaveAttribute(`href`, `/local`)
    expect(link).not.toHaveAttribute(`target`)
  })
})
