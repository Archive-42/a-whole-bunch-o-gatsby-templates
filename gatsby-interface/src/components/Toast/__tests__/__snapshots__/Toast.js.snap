// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast renders unchanged 1`] = `
<div>
  <div
    class="css-6pdnk1-toastCss-Toast-Toast"
    data-reach-alert="true"
    data-testid="toast"
  >
    <span
      class="css-3wzd5o-statusCss-Toast-Toast"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
        />
      </svg>
    </span>
    <div
      class="css-1e1jz1t-messageCss"
    >
      Lorem ipsum
    </div>
    <button
      aria-label="close"
      class="css-1o0uy6d-closeButtonCss"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </button>
  </div>
  <div
    class="css-j6docj-toastCss-Toast-Toast"
    data-reach-alert="true"
    data-testid="toast"
  >
    <span
      class="css-18fmh0q-statusCss-Toast-Toast"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
        />
      </svg>
    </span>
    <div
      class="css-1e1jz1t-messageCss"
    >
      Lorem ipsum
    </div>
    <button
      aria-label="close"
      class="css-1o0uy6d-closeButtonCss"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </button>
  </div>
</div>
`;
